samcli/hook_packages/terraform/hooks/prepare/types.py

```diff
@@ -129,12 +131,19 @@ def collect(self, properties: ResourceTranslationProperties):
         """
         Collect any properties required for future resource processing after traversing the Terraform modules.
 
+        In parent class we collect all the required properties. This logic can be customized in the children classes.
+
         Parameters
         ----------
         properties: ResourceTranslationProperties
             Properties acquired specific to an AWS resource when iterating through a Terraform module
         """
-        raise NotImplementedError
+        resolved_config_address = get_configuration_address(properties.resource_full_address)
```

Comment:
Was just about to raise a PR doing the same thing, beat me to it 😄 

Should we also remove the implementations of `Collect` from Lambda function and Layer and let them inherit from the parent?

Reply:
yes we can do that. We need it for Lambda function any way, as it will be destination resource in one of our mappings

Reply:
done